### ISSUE TITLE
Destructuring exported array with `require` causes error

### ISSUE THREAD

**Issue by lumaxis** [microsoft/TypeScript#44658]
# Bug Report

I'm progressively adding TypeScript to an existing JS codebase. In several of our files, there is an existing pattern of exporting arrays like so: `module.exports = [A, serializeA]`.
When requiring and destructuring this export like `const [A] = require('./a')`, I get an unexpected error `Module '"./a"' has no exported member 'A'. ts(2305)` which I believe to be a bug.

### 🔎 Search Terms

module has no exported member TS2305 

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about `module.exports`

### ⏯ Playground Link

Not sure how to repro this in a Playground work Workbench since I believe it requires files and exports/requries.

### 💻 Code

```js
// a.js
class A {}
module.exports = [A]

// b.js
const [A] = require('./a') // Module '"./a"' has no exported member 'A'. ts(2305)
```

### 🙁 Actual behavior

The compiler shows an error

### 🙂 Expected behavior

The compiler shows no error and the exported array is correctly destructured and the type infered.

###

**Comment by RyanCavanaugh** [comment]
Seems like we missed a case in the destructuring logic where we forgot to check whether it's an array or an object?

**Comment by assapir** [comment]
I would like to take a look at this, can I have some direction on where to start looking at it?

**Comment by MarcosAtApprentice** [comment]
Just ran into this issue, very tricky to figure out the source of this. Nothing was working for type inference, until I tried first assigning the result of `require()` into a variable and only then destructuring that variable.

Using latest version:

![image](https://user-images.githubusercontent.com/84727903/127036470-7bbf8285-40a8-4509-b3af-29031ab4caf3.png)

This works:

![image](https://user-images.githubusercontent.com/84727903/127036507-b7ddf7fa-f81d-41be-a7f6-8740a3116318.png)

but this doesn't work:

![image](https://user-images.githubusercontent.com/84727903/127036525-46b53eac-a8b9-4b98-a074-63cb5b3ce69a.png)

Having to require() into a variable, then destructuring that variable for it work doesn't seem like intended behavior.

I also found someone having the same issue who found the same workaround in [this Stack Overflow thread](https://stackoverflow.com/questions/64266318/how-do-you-add-a-jsdoc-type-to-a-destructured-variable-vscode).
